### DISCUSSION TITLE
Migrations are not in order with the related table

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo#615] [General]
Independent table should be created before the dependent table, but this is not the case with piccolo-migration.

**Comment by dantownsend** [comment]
We use Python's `graphlib` to sort the tables based on their foreign keys:

https://github.com/piccolo-orm/piccolo/blob/0b9a86b3c304dccd912115c2d3d77c4bebc132bc/piccolo/table.py#L1381

However, there might be situations where it can't sort them properly. Is your schema quite complex?

**Reply by sumitsharansatsangi** [reply]
Yeah, five classes are sequentially dependent on other.
Like as follows:-
Table1 -> Table2 -> Table3 -> Table4 -> Table5 .

For simple dependency , it works fine.
Like as follows 
Table1 -> Table2

**Reply by dantownsend** [reply]
You're right - this is a bug.

```python
from piccolo.columns import Varchar, ForeignKey
from piccolo.table import Table

class Manager(Table):
    name = Varchar()

class Manager2(Table):
    fk = ForeignKey(Manager)

class Manager3(Table):
    fk = ForeignKey(Manager2)

class Manager4(Table):
    fk = ForeignKey(Manager3)

# If I add the fifth it breaks:
# class Manager5(Table):
#     fk = ForeignKey(Manager4)
```

For some reason if there are 5 or more tables in the chain it fails 🤔 

The work around for now is to create two smaller migrations. I need to investigate why this is failing.

**Reply by dantownsend** [reply]
It's not a problem with `sort_table_classes`:

```python 
In [5]: sort_table_classes([Manager2, Manager5, Manager4, Manager3, Manager])
Out[5]: [Manager, Manager2, Manager3, Manager4, Manager5]
```

Must be something wrong within `MigrationManager`.

**Reply by dantownsend** [reply]
Hmm, actually it is a bug with `sorted_table_classes`. Sometimes it gives the wrong order.

**Reply by dantownsend** [reply]
I've figured out the problem - will push a fix later today.

**Reply by dantownsend** [reply]
This should be fixed now - if you try `piccolo==0.90.0`

**Reply by sumitsharansatsangi** [reply]
Thanks a lot, @dantownsend,  for the fix.

**Reply by sumitsharansatsangi** [reply]
@dantownsend , @sinisaos , Please visit [this](https://github.com/piccolo-orm/piccolo_admin/discussions/220).
The discussion I raised here is related to this.

**Reply by sumitsharansatsangi** [reply]
I am not able to add my data from piccolo admin to the table that was related like as follows
TableA-> TableB-> TableC -> TableD -> TableE.

Meanwhile, I am trying to do it with piccolo-0.90.0
